pkg/model/estimate.go

```diff
@@ -3,165 +3,81 @@ package model
-	var power float64
-	if metricIndex >= 0 && totalUsage > 0 {
-		power = podMetricValue[metricIndex] / totalUsage * float64(totalPower)
-	} else {
-		power = float64(totalPower) / podNumber
-	}
-	return uint64(math.Ceil(power))
-}
-
-func GetPowerFromUsageRatio(podMetricValues [][]float64, totalCorePower, totalDRAMPower, totalUncorePower, totalPkgPower uint64, sumUsage map[string]float64) (podCore, podDRAM, podUncore, podPkg []uint64) {
-	podNumber := float64(len(podMetricValues))
-	totalCoreUsage := sumUsage[config.CoreUsageMetric]
-	totalDRAMUsage := sumUsage[config.DRAMUsageMetric]
-	totalUncoreUsage := sumUsage[config.UncoreUsageMetric]
-	totalUsage := sumUsage[config.GeneralUsageMetric]
-
-	// Package (PKG) domain measures the energy consumption of the entire socket, including the consumption of all the cores, integrated graphics and
-	// also the "unknown" components such as last level caches and memory controllers
-	pkgUnknownValue := totalPkgPower - totalCorePower - totalUncorePower
-
-	// find ratio power
-	for _, podMetricValue := range podMetricValues {
-		coreValue := getRatio(podMetricValue, coreMetricIndex, totalCoreUsage, totalCorePower, podNumber)
-		dramValue := getRatio(podMetricValue, dramMetricIndex, totalDRAMUsage, totalDRAMPower, podNumber)
-		uncoreValue := getRatio(podMetricValue, uncoreMetricIndex, totalUncoreUsage, totalUncorePower, podNumber)
-		unknownValue := getRatio(podMetricValue, generalMetricIndex, totalUsage, pkgUnknownValue, podNumber)
-		pkgValue := coreValue + uncoreValue + unknownValue
-		podCore = append(podCore, coreValue)
-		podDRAM = append(podDRAM, dramValue)
-		podUncore = append(podUncore, uncoreValue)
-		podPkg = append(podPkg, pkgValue)
-	}
-	return
+type ComponentPowerResponse struct {
+	Powers  map[string][]float64 `json:"powers"`
+	Message string               `json:"msg"`
 }
 
-func GetDynamicPower(metricNames []string, podMetricValues [][]float64, corePower, dramPower, uncorePower, pkgPower, gpuPower []float64) []float64 {
+func GetPowerByEstimator(metricNames []string, metricValues [][]float64, outputType ModelOutputType, systemFeatures, systemValues []string, powerResponse interface{}) {
+	outputTypeStr := outputType.String()
 	powerRequest := PowerRequest{
-		ModelName:       config.EstimatorModel,
-		MetricNames:     metricNames,
-		PodMetricValues: podMetricValues,
-		CorePower:       corePower,
-		DRAMPower:       dramPower,
-		UncorePower:     uncorePower,
-		PkgPower:        pkgPower,
-		GPUPower:        gpuPower,
-		SelectFilter:    config.EstimatorSelectFilter,
+		ModelName:      config.SelectedModel[outputTypeStr],
+		MetricNames:    metricNames,
+		MetricValues:   metricValues,
+		SelectFilter:   config.ModelSelectFilter,
+		OutputType:     outputTypeStr,
+		SystemFeatures: systemFeatures,
+		SystemValues:   systemValues,
 	}
 	powerRequestJSON, err := json.Marshal(powerRequest)
 	if err != nil {
 		log.Printf("marshal error: %v (%v)", err, powerRequest)
```

Comment:
We have changed all the logs to use klog with verbosity support #238

The general rule of thumb is:
- verbosity V(1): must always be printed
- verbosity V(3): extra info
- verbosity V(4): only debug info
- verbosity V(5): only extra debug info